app/templates/users/details.html

```diff
@@ -1,91 +1,107 @@
+      <thead>
+        <tr>
+          <th>App name</th>
+          <th>Admin</th>
+          {% if signedInUser %}
+            <td>&nbsp;</td>
+          {% endif %}
+        </tr>
+      </thead>
+      <tbody>
+        {% for app in user.apps %}
           <tr>
-            <th>App name</th>
-            <th>Admin</th>
+            <td>
+              <a href="/apps/{{ app.id }}">{{ app.name }}</a>
+            </td>
+            <td>
+              {% if app.role == 0 %}
+                Yes
+              {% else %}
+                No
+              {% endif %}
+            </td>
             {% if signedInUser %}
-              <td>&nbsp;</td>
-            {% endif %}
-          </tr>
-        </thead>
-        <tbody>
-          {% for app in user.apps %}
-            <tr>
-              <td>
-                <a href="/apps/{{ app.id }}">{{ app.name }}</a>
-              </td>
-              <td>
+              <td class="align-right">
                 {% if app.role == 0 %}
-                  Yes
-                {% else %}
-                  No
+                  <a class="button button-secondary" href="{{ url_for('apps.details', {id: app.id}) }}">Manage app</a>
+                  <a class="button button-warning js-confirm" href="/apps/delete/{{ app.id }}/{{ user.id }}">Delete app</a>
                 {% endif %}
               </td>
-              {% if signedInUser %}
-                <td class="align-right">
-                  {% if app.role == 0 %}
-                    <a class="button button-secondary" href="/apps/manage/{{ app.id }}">Manage app</a>
-                    <a class="button button-secondary js-confirm" href="/apps/delete/{{ app.id }}/{{ user.id }}">Delete app</a>
-                  {% endif %}
-                </td>
-              {% endif %}
-            </tr>
-          {% endfor %}
-        </tbody>
-      </table>
-    {% else %}
-      <p>None</p>
-    {% endif %}
-
-    {% if signedInUser %}
-      <p><a class="button button-secondary" href="/apps/new">Create new app</a></p>
-    {% endif %}
-  </div>
-
-
-  <h2>{{ list_prefix }} app data sources</h2>
-  {% if user.datasources.length %}
-    <ul class="list list-bullet">
-      {% for datasource in user.datasources %}
-        <li><a href="/datasources/show/{{ datasource.id }}">{{ datasource.bucket_name }}</a></li>
-      {% endfor %}
-    </ul>
+            {% endif %}
+          </tr>
+        {% endfor %}
+      </tbody>
+    </table>
   {% else %}
     <p>None</p>
   {% endif %}
 
-</main>
+  {% if signedInUser %}
+    <p><a class="button button-secondary" href="/apps/new">Create new app</a></p>
+  {% endif %}
+</div>
+
+
+<h2>User's app data sources</h2>
+{% if user.users3buckets.length %}
+  <table class="form-group">
+    <thead>
+      <tr>
+        <th>Name</th>
+        <th>Admin</th>
+        <th>Read/write access</th>
+        <td><span class="visuallyhidden">Actions</span></td>
+      </tr>
+    </thead>
+    {% for bucket in user.users3buckets %}
```

Comment:
You may want to rename this variable to `users3bucket` to avoid confusion as this is a loop on the access objects, not the actual buckets.